react-app/src/library/services/headset.ts

```diff
@@ -0,0 +1,238 @@
+import JabraService from './vendor-implementations/jabra/jabra';
+import JabraNativeService from './vendor-implementations/jabra/jabra-native/jabra-native';
+import ApplicationService from './application';
+import { ConsumedHeadsetEvents } from '../types/consumed-headset-events';
+import { CallInfo } from '../types/call-info';
+import { EventInfo, VendorConversationIdEvent, VendorEvent, HoldEventInfo, MutedEventInfo } from '../types/emitted-headset-events';
+export default class HeadsetService {
+  private static instance: HeadsetService;
+
+  plantronics: VendorImplementation;
+  jabraNative: VendorImplementation;
+  jabra: VendorImplementation;
+  sennheiser: VendorImplementation;
+  application: ApplicationService;
+  selectedImplementation: VendorImplementation;
+  headsetEvents$: Observable<ConsumedHeadsetEvents>;
+
+  private _headsetEvents$: BehaviorSubject<ConsumedHeadsetEvents>;
+  private _implementations: VendorImplementation[] = [];
+  private logger: any;
+
+  private constructor(config: ImplementationConfig) {
+    // super();
+    //TODO: Just a temporary typing for testing purposes
+    this._headsetEvents$ = new BehaviorSubject<ConsumedHeadsetEvents | any>({
+      event: '',
+      payload: {}
+    });
+    this.headsetEvents$ = this._headsetEvents$.asObservable();
+
+    this.application = ApplicationService.getInstance();
+    this.selectedImplementation = this.implementations[0]; // Using the first just because it's the first
+    this.logger = config?.logger || console;
+    this.plantronics = PlantronicsService.getInstance({ logger: this.logger });
+    this.jabraNative = JabraNativeService.getInstance({ logger: this.logger });
+    this.jabra = JabraService.getInstance({ logger: this.logger });
+    this.sennheiser = SennheiserService.getInstance({ logger: this.logger });
+
+    [this.plantronics, this.jabraNative, this.sennheiser, this.jabra]
+      .forEach(implementation => this.subscribeToHeadsetEvents(implementation));
+  }
+
+  static getInstance(config: ImplementationConfig): HeadsetService {
+    if (!HeadsetService.instance) {
+      HeadsetService.instance = new HeadsetService(config);
+    }
+
+    return HeadsetService.instance;
+  }
+
+  get implementations(): VendorImplementation[] {
+    if (this._implementations.find((implementation) => implementation)) {
+      return this._implementations;
+    }
+
+    const implementations: VendorImplementation[] = [];
+    if (this.application.hostedContext.supportsJabra()) {
+      implementations.push(
+        // this.application.hostedContext.isHosted() ? this.jabraNative : this.jabraChrome
+        this.application.hostedContext.isHosted() ? this.jabraNative : this.jabra
+        // this.jabra
+      );
+    }
+    implementations.push(this.plantronics);
+    implementations.push(this.sennheiser);
+
+    this._implementations = implementations;
+    return this._implementations;
+  }
+
+  getHeadSetEventsSubject = (): BehaviorSubject<ConsumedHeadsetEvents> => {
```

Comment:
We don't need this right? The probably shouldn't be exposed.

Reply:
This was originally added in for testing purposes but it does help with a weird typing issue I was seeing after being consumed by SDK. Will work on this